tests/scripts/translate_ciphers.py

```diff
@@ -0,0 +1,584 @@
+#!/usr/bin/env python3
+
+# translate_ciphers.py
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Translate ciphersuite names in MBedTLS format to OpenSSL and GNUTLS
+standards.
+
+To test the translation functions run:
+python3 -m unittest translate_cipher.py
+"""
+
+import re
+import argparse
+import unittest
+
+class TestTranslateCiphers(unittest.TestCase):
```

Comment:
I am not sure to understand here. With the testing of the translate functions we are still maintaining a list of ciphersuite names in the thre libraries thus why not using it directly ?

Reply:
Fair point, I could just look it up in the list which might be easier...
I guess you could make the argument that you wouldn't need to explicitly define new cipher suite names for all conventions as they should follow the same rules, but I guess that it would be bad practice not to verify it
@mpg what do you think? Should I just use the list defined in the test, and remove the tests?
also issue #4674 states:
> Validation: the PR should include evidence that the new script translates correctly by comparing its results to the existing lists.
so if there is a check, but also translating functions, maybe there's another reason I am not sure about?

Reply:
I don't think we should just look up translations in a table. Having functions that do the translation has value: it spells out the rules. That way, when we add a new ciphersuite we don't have to reverse-engineer the OpenSSL naming scheme by looking at various places in the existing list to tediously guess what the new entries should be. (Said otherwise, having a table listing all translations in the python script would just move the problem from `compat.sh` to another place.)

Taking a step back: initially, the plan was not to keep the list after this PR. The task description said the PR should provide some evidence that the new script works correctly, without any details about what form that evidence might take. Joe had written some tests scripts, and initially it was unclear whether they would be temporary or not. Then Gilles suggested we keep the tests. But as Ronald points out, keeping the tests in their current form means still having the full list, and gives the impression that we may have to maintain that list next time we add new ciphersuites...

I suggest keeping the test but with a reduced list: we don't need to have every single ciphersuite that we support listed here. Just enough to make sure each case in the translation function is exercised at least once (that is, maintain full coverage). That way, we keep the security of the tests (if we evolve the script, we'll be confident we don't break things) without still having the full list around. If we add ciphersuites in the future, if the existing rules work for them we don't have to add anything to the list, and otherwise we'll add code in the translation function and a corresponding entry in the list to ensure that code is tested.

Wdyt?

Reply:
I think that sounds like the most sensible suggestion. If we are trimming down the list, I guess I should only choose a few ciphers that fullfill each check/step in the translation function? Trimming the list would definitely give the impression that it is purely there to test the functionality, but I may need advice on what is appropriate to remove.

Reply:
I think @mpg suggestion is a very good suggestion as well.